probe-rs/src/probe.rs

```diff
@@ -374,7 +366,7 @@ impl Probe {
         &mut self,
         scan_chain: Vec<ScanChainElement>,
     ) -> Result<(), DebugProbeError> {
-        if self.attached {
+        if !self.attached {
```

Comment:
Almost forgot the most important change, the reason for this entire PR :sweat_smile:. This behaviour is now flipped so you set the chain before attaching to the target, it made no sense to do it the other way around before and only worked because of the inconsistencies this PR fixes.